code/datums/components/tilted.dm

```diff
@@ -0,0 +1,72 @@
+/**
+ * A component that should be attached to things that have been tilted over, and can be righted.
+ * This can optionally block normal attack_hand interactions
+ */
+
+/datum/component/tilted
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+	/// How long it should take to untilt
+	var/untilt_duration
+	/// Whether we should block any interactions with it
+	var/block_interactions
+	/// The angle by which we rotated as a result of tilting. Should help us avoid cases where something gets tilted until it's upright.
+	var/rotated_angle
+
+/datum/component/tilted/Initialize(_untilt_duration = 16 SECONDS, _block_interactions = FALSE, _rotated_angle)
+	. = ..()
+	untilt_duration = _untilt_duration
+	block_interactions = _block_interactions
+	rotated_angle = _rotated_angle
```

Comment:
Not all AltClick  procs call . = ..(), I think this might need some kind of COMPONENT_INCOMPATIBLE if you can figure something out.

Reply:
I don't think I can really just check to see if anything has an override for AltClick, altclick might just need a SHOULD_CALL_PARENT after this PR goes in